File: backend/routers/feed.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session
from database import get_db
from schemas.reel import ReelResponse
from services.feed import get_global_feed, get_feed_by_genre

router = APIRouter(prefix="/feeds", tags=["Feeds"])


@router.get("/", response_model=list[ReelResponse])
def global_feed(
    skip: int = 0,
    limit: int = 10,
    db: Session = Depends(get_db)
):
    return get_global_feed(db, skip, limit)


@router.get("/genre/{genre}", response_model=list[ReelResponse])
def genre_feed(
    genre: str,
    skip: int = 0,
    limit: int = 10,
    db: Session = Depends(get_db)
):
    return get_feed_by_genre(db, genre, skip, limit)

File: frontend/src/api/feed.ts
import api from './axios'
import type { Reel } from '../types'

export async function getGlobalFeed(skip = 0, limit = 10): Promise<Reel[]> {
  const res = await api.get('/feeds/', { params: { skip, limit } })
  return res.data
}

export async function getGenreFeed(genre: string, skip = 0, limit = 10): Promise<Reel[]> {
  const res = await api.get(`/feeds/genre/${encodeURIComponent(genre)}`, { params: { skip, limit } })
  return res.data
}
